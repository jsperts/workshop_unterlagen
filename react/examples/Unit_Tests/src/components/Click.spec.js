import React from 'react';
import { shallow } from 'enzyme';

import Click from './Click';

describe('Click', () => {
  test('Should call onIncrement when the button is clicked', () => {
    const onIncrement = jest.fn();

    const wrapper = shallow(<Click onIncrement={onIncrement} />);

    wrapper.find('button').simulate('click');

    expect(onIncrement).toHaveBeenCalled();
  });

  test('Should match the snapshot', () => {
    const wrapper = shallow(<Click />);
    expect(wrapper.getNodes()).toMatchSnapshot();
  });
});
