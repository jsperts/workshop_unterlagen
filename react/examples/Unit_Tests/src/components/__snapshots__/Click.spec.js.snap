// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click Should match the snapshot 1`] = `
Array [
  <div>
    <p>
        Aktuelle Anzahl: 
    </p>
    <button
        onClick={undefined}
        type="button"
    >
        Klick
    </button>
</div>,
]
`;
